New Shapefile Layer: reorganise additional dimensions checkboxes

agiudiceandrea opened the pull request:
## Description
In the New Shapefile Layer dialog there are two checkboxes ("Include Z dimension", "Include M values") that allow to create shapefiles with added Z and M "additional dimensions"

![image](https://user-images.githubusercontent.com/16253859/52096572-a5afeb00-25c7-11e9-9fc2-da38e1c103b5.png)

Since the two checkboxes are not non mutually exclusive, the user could check none, only one, or both of them, thus wrongly thinking that 4 kinds of new shapefile layer could be created, e.g. for Point geometry type:
- Point (X,Y)
- PointZ (X,Y,Z)
- PointM (X,Y,M)
- PointZM (X,Y,Z,M)

while actually only 3 types of Point shapefiles do exists:
- Point (X,Y)
- PointM (X,Y,M)
- PointZ = PointZM (X,Y,Z,M)

In fact, whether checking only the "Include Z dimension" checkbox or checking both the "Include Z dimension" and the "Include M values" checkboxes, will lead to create the very same ZM new shapefile layer.

In order to solve this ambiguity, I propose to change the New Shapefile Layer dialog adding the label "Additional dimensions" in the checkboxes row and changing the two non mutually exclusive in three mutually exclusive checkboxes "None", "Z + M values", "M values", with "None" pre checked by default (since it seems to me it's not possible to have two mutually exclusive checkboxes that could be both unchecked after one of them is chechekd).

![image](https://user-images.githubusercontent.com/16253859/52111112-0577b780-2603-11e9-87bb-a1515ecc4ae2.png)

Note: this proposed change however will make the New Shapefile Layer dialog inconsistent with the other New SpatiaLite/GeoPackage/Memory layer dialogs about the Z and M checkboxes. Could this be a problem?

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

nyalldawson commented:
This change seems reasonable to me. But please use radio buttons for mutually exclusive options instead of checkboxes

agiudiceandrea committed:
New Shapefile Layer: reorganise additional dimensions checkboxes

because shapefiles could only be
- plain X,Y
- with M values: X,Y,M
- with Z and M values: X,Y,Z,M

nirvn commented:
An alternative approach to try and keep UI consistency across new layer dialogs would be to keep both checkboxes, and if the user toggles Z on, the M checkbox should automatically be toggled on and its state set to disabled.

agiudiceandrea commented:
> This change seems reasonable to me. But please use radio buttons for mutually exclusive options instead of checkboxes

new version with radio buttons
![image](https://user-images.githubusercontent.com/16253859/52124588-f8ba8a00-2629-11e9-90d9-c16fdf6f319f.png)

agiudiceandrea commented:
> An alternative approach...

@nirvn, I do like your idea! Are you going to implement it?

nyalldawson commented:
Honestly, I prefer the explicit radio buttons over checkboxes which automatically turn on and off. Radio buttons make the logic obvious to users: "these are the 3 options you get"

nyalldawson commented:
I'm happy with this current PR. @nirvn?

m-kuhn commented:
I also like the current state.

nirvn commented:
The majority wins :) +1 to merge.
